assets/pages/new-contributions-landing/contributionsLandingActions.js

```diff
@@ -144,7 +163,21 @@ const makeOneOffPaymentData: (PaymentAuthorisation, State) => PaymentFields = (t
   },
 });
 
-const makeRegularPaymentData: (PaymentAuthorisation, State) => PaymentFields = (token, state) => ({
+function paymentDetailsFromAuthorisation(authorisation: PaymentAuthorisation): PaymentDetails {
+  switch (authorisation.paymentMethod) {
+    case 'Stripe': return { stripeToken: authorisation.token };
+    case 'PayPal': return { baid: authorisation.token };
+    case 'DirectDebit': return {
+      accountHolderName: authorisation.accountHolderName,
+      sortCode: authorisation.sortCode,
+      accountNumber: authorisation.accountNumber,
+    };
+    // TODO: what is a sane way to handle such cases?
+    default: throw new Error('If Flow works, this cannot happen');
```

Comment:
I've been using the fall-through method as above (with Stripe as the default), as it is still very easy to get around Flow–although it would require a dev going above and beyond 🤡